tests/test_pds4_create_xml_index.py

```diff
@@ -0,0 +1,460 @@
+    label_file = 'tester_label_1.xml'
+    schema_files = tools.process_schema_location(test_files_dir / label_file)
+    assert schema_files[0] == 'https://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1B00.xsd'
+    assert schema_files[1] == 'https://pds.nasa.gov/pds4/disp/v1/PDS4_DISP_1B00.xsd'
+    assert schema_files[2] == 'https://pds.nasa.gov/pds4/mission/cassini/v1/PDS4_CASSINI_1B00_1300.xsd'
+
+
+# Set parameters values that you would like to pass into test_elements_file.
+# In this case, we are running the same test with different sets of
+# golden_file, new_file, and cmd_line.
+    
+# Currently, element_file unit tests are commented out due to the current code not
+# creating intended results.
+@pytest.mark.parametrize(
+        'golden_file,new_file,cmd_line',
+        [
+            # simple test cases
+            (str(test_files_dir / 'elements_file_success.csv'),
+             'elements_file.csv',
+             [
+                str(test_files_dir),
+                'tester_label_1.xml',
+                '--elements-file',
+                str(root_dir / 'samples/sample_elements.txt'),  
+                '--output-file',     
+             ]
+            ),
+            # Okay, basic capture
+            (str(test_files_dir / 'elements_file_success_2.csv'),
+             'elements_file_2.csv',
+             [
+                str(test_files_dir),
+                'tester_label_2.xml',
+                '--elements-file',
+                str(root_dir / 'samples/element_2.txt'),  
+                '--output-file',     
+             ]
+            ),
+            # Multiple partial captures
+            (str(test_files_dir / 'elements_file_success_3.csv'),
+             'elements_file_3.csv',
+             [
+                str(test_files_dir),
+                'tester_label_2.xml',
+                'tester_label_3.xml',
+                '--elements-file',
+                str(root_dir / 'samples/element_3.txt'),  
+                '--output-file',     
+             ]
+            ),
+
+            # exclusion test
+            (str(test_files_dir / 'elements_file_success_4.csv'),
+            'elements_file_4.csv',
+             [
+                str(test_files_dir),
+                'tester_label_*.xml',
+                '--elements-file',
+                str(root_dir / 'samples/element_4.txt'),  
+                '--output-file',     
+             ]
+            )
+
+
+        ]
+                        )
+                    
+def test_elements_file(golden_file, new_file, cmd_line):
+    # Create a temporary directory in the same location as the test_files directory
+    with tempfile.TemporaryDirectory(dir=test_files_dir.parent) as temp_dir:
+        temp_dir_path = Path(temp_dir)
+        
+        # THE PATH TO THE NEW FILE
+        path_to_file = temp_dir_path / new_file
+        # Call main() function with the simulated command line arguments
+        cmd_line.append(str(path_to_file))
```

Comment:
Mostly this is from the original example Dave provided. However, now that we're using `--dump-available-elements` in some of the unit tests, there can be either a `csv` or a `txt` temp file. We could:
- keep it as is
- Change all the successful files to one file type.

Reply:
I'll leave that decision up to you. Having a single output filename (probably `.txt`) seems like it would make the code 
simpler, but it's not a big deal either way.

Reply:
Keeping it as is, then.